src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -572,6 +481,131 @@ public static List<String> getJacocoAgentOptions() {
+                        if (snapshotPlugins.contains(line)) {
+                            continue;
+                        }
+                        final URL url = new URL(index, line + ".jpi");
+                        File f;
+                        try {
+                            f = new File(url.toURI());
+                        } catch (IllegalArgumentException x) {
+                            if (x.getMessage().equals("URI is not hierarchical")) {
+                                throw new IOException(
+                                        "You are probably trying to load plugins from within a jarfile (not possible). If" +
+                                                " you are running this in your IDE and see this message, it is likely" +
+                                                " that you have a clean target directory. Try running 'mvn test-compile' " +
+                                                "from the command line (once only), which will copy the required plugins " +
+                                                "into target/test-classes/test-dependencies - then retry your test", x);
+                            } else {
+                                throw new IOException(index + " contains bogus line " + line, x);
+                            }
+                        }
+                        if (f.exists()) {
+                            FileUtils.copyURLToFile(url, new File(plugins, line + ".jpi"));
+                        } else {
+                            FileUtils.copyURLToFile(new URL(index, line + ".hpi"), new File(plugins, line + ".jpi"));
+                        }
+                    }
+                }
+            }
+        }
+        for (String extraPlugin : extraPlugins) {
+            URL url = RealJenkinsRule.class.getClassLoader().getResource(extraPlugin);
+            String name;
+            try (InputStream is = url.openStream(); JarInputStream jis = new JarInputStream(is)) {
+                Manifest man = jis.getManifest();
+                if (man == null) {
+                    throw new IOException("No manifest found in " + extraPlugin);
+                }
+                name = man.getMainAttributes().getValue("Short-Name");
+                if (name == null) {
+                    throw new IOException("No Short-Name found in " + extraPlugin);
+                }
+            }
+            FileUtils.copyURLToFile(url, new File(plugins, name + ".jpi"));
+        }
+        System.out.println("Will load plugins: " + Stream.of(plugins.list()).filter(n -> n.matches(".+[.][hj]p[il]")).sorted().collect(Collectors.joining(" ")));
+    }
+
+    /**
+     * Deletes {@code JENKINS_HOME}.
+     *
+     * This method does not need to be invoked when using {@code @Rule} or {@code @ClassRule} to run {@code RealJenkinsRule}.
+     */
+    public void deprovision() throws Exception {
+        tmp.dispose();
+        home.set(null);
```

Comment:
This line is new, and I think it only matters for dynamically-provisioned instances. In normal cases this happens after the test method completes.